File: Native/Native.cpp
#include "Native.h"
#include <stdio.h>
Native::Native()
{
}
Native::~Native()
{
}
void Native::Init(OutGoingCallback onNotify)
{
	_onNotify = onNotify;
	OutGoingMessage("Cpp UnManaged class Init Complete");
}
void Native::OutGoingMessage(std::string outGoingMessage)
{
	//printf("Cpp UnManaged class -> Managed class -> C# Outgoing Message\n");
	_onNotify(outGoingMessage);
}
void Native::InCommingMessage(std::string inGoingMessage)
{
	//printf("C#-> Managed class -> Cpp UnManaged class ->  ingoing Message\n");
	printf("[Cpp] Notify InCommingMessage :: %s\n", inGoingMessage.c_str());
}
void Native::Stop()
{
	printf("Cpp Native Stop\n");
}

File: CSNative/Native.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NativeCS
{
    public class Native : CS.CSInterface
    {
        public event Action<object, string> OnNotify;
        private bool _disposed;
        private void Dispose(bool isDispose)
        {
            _disposed = isDispose;
        }
        public void Dispose()
        {
            if (_disposed)
                return;
            Dispose(true);
        }

        public void Init()
        {
            OutGoinMessage("C# Init Complete");
        }
        public void Run()
        {
            
        }
        public void InCommingMessage(string message)
        {
            Console.WriteLine($"[C#] Notify InCommingMessage :: {message}");
        }
        void OutGoinMessage(string message)
        {
            //Console.WriteLine("C# OutGoinMessage");
            OnNotify?.Invoke(this, message);
        }
        public void Stop()
        {
            Console.WriteLine("Stop C# Native Class");
        }
    }
}
